### ISSUE TITLE
Scala-Native Support (1000USD Bounty)

### ISSUE THREAD

**Issue by lihaoyi** [com-lihaoyi/requests-scala#156]
[STTP](https://github.com/softwaremill/sttp) Support Scala-Native via Curl. Requests-Scala should be able to support it via the same approach. We just need it to be wired into the Requests-Scala codebase, possibly refactoring out the actual making of the request into `src-jvm/` and `src-native/` platform-specific code. 

Any platform-specific functionality that doesn't work on Scala Native should be called out, explained clearly, and their associated tests moved into platform-specific folders as well

To incentivize contribution, I'm putting a 1000USD bounty on resolving this ticket. This is payable via bank transfer, and at my discretion in case of ambiguity.

**Comment by domaspoliakas** [comment]
I'll give this one a go 🙂

**Comment by domaspoliakas** [comment]
Just wanted to mention that I am still working on this and have not abandoned it! I have PRed `HttpCookie` to scala native now https://github.com/scala-native/scala-native/pull/3927 and I'll be looking at how to solve the issue of `SSLContext` next.

**Comment by domaspoliakas** [comment]
A wee little update - the approach that sttp has taken unfortunately does not seem viable here; sttp simply does not allow any streaming with the curl backend which is how they got away with only using the libcurl `easy` API, but given that this lib is based on [`InputStream` at its core](https://github.com/com-lihaoyi/requests-scala/blob/master/requests/src/requests/Requester.scala#L165) - my understanding is that this is not quite possible. That is - curl's easy API does not allow piecemeal consumption of the data, and therefore we'd have to buffer the whole response into memory if we were to use the same approach. However curl's `multi` API I think has all the pieces that are needed here. There's [http4s-curl](https://github.com/http4s/http4s-curl) as prior art which does exactly that, so I'm trying to see if I can adapt a similar approach for this repo.

**Comment by lihaoyi** [comment]
@domaspoliakas sounds good!

**Comment by lqhuang** [comment]
I have subscribed to this issue for a long time due to I also want to give a try in the beginning. But I'm a little worry about my junior skills in Scala Naive. Thanks for @domaspoliakas taking over the ticket.

I still hope I could give a hand, or we could cooperate to solve some problems?

Like above mentioned, we're currently trying to build Scala native API on top of `libcurl` just like `https-curl` did. I'm thinking could we extract the core `libcurl` as an individual binding lib, so both `https-curl` and `requests` are able to create their user interface on it?

Someone has tried to create a binding (https://github.com/fsat/scala-native-libcurl-bindings) years ago and obviously abandoned already.

What's your opinions?

Regards

**Comment by domaspoliakas** [comment]
I personally do like the idea (and indeed you could add sttp to the list of libs that could benefit from such a thing here), but one downside is that it would add a dependency. This has a couple of drawbacks. The first is that requests-scala is a very minimalist library and practically does not add any transitive dependencies, which this would change a little. The second is that it would be a dependency outside of this organisation's control. Whether these are blockers or not is for @lihaoyi to decide.

If you do get the green light - take a look at https://github.com/domaspoliakas/requests-scala/blob/scala-native-cross-build/requests/src-native/requests/curl.scala . I've already covered a chunk of the API, at least the things that I needed so far, which you could use as a starting point. I used sttp as a reference for that, so you'll notice that the style of the bindings is in many ways reminiscent of their implementaiton. Another thing to mention is that I of course treated this as an internal detail of requests-scala so it may need some cleanup in order to be published as a library of its own.

**Comment by lihaoyi** [comment]
I dont mind a third party dep as along as it is well maintained and stable. Most important thing is that the end to end workflow works; we can refactor the implementation however we like later

**Comment by sideeffffect** [comment]
You could potentially use curl bindings automatically generated by sn-bindgen
https://github.com/indoorvivants/sn-bindgen-examples by @keynmol

**Comment by keynmol** [comment]
Note that sn-bindgen produces Scala3-only bindings.
My recommendation for this would be for libraries to ship their own handcrafted bindings as long as the API surface is small.

Alternatively, lots of binding code can be copypasted from https://github.com/indoorvivants/sn-bindgen-examples/blob/main/example-curl/src/main/scala/generated/curl.scala#L3113 and adapted to fit Scala 2. It should be a lot easier than handcrafting the entire API

**Comment by domaspoliakas** [comment]
The scala-3 requirement was the main reason for hand-crafting the bindings in this case, although I have used the generated bindings as a guide for some of the trickier parts of the API. The API is actually fairly small in the end, and I don't think I'll be adding anything to the bindings at this point. 

I'll take this opportunity to mention that parts of the test suite work already, so there is light at the end of the tunnel. There's still a bit of work before all of it is ok (the looming threat is still `SSLContext` and however I'll need to solve that), and then I'll need to clean it all up, but the end is definitely getting nearer :)

**Comment by lihaoyi** [comment]
Sounds great @domaspoliakas !

**Comment by lihaoyi** [comment]
@domaspoliakas any updates here? If not I'll be putting it up in the next set of bounties

**Comment by domaspoliakas** [comment]
June was unfortunately a tough month to find time for this and I apologise for disappearing as I have. I have been working on this again as of last weekend however. Simple functionality is working at the moment, but there are still quite a few things left:

- SSL related stuff
- Concurrent transfers (single-threaded)
- Multithreading
- Redirect handling
- Cookie handling
- A bug in multipart uploads

That last one has eaten quite a lot of my time so far with unfortunately not much to show for it. Generally uploads seem to work (some hand testing and some of the test suite corroborate this), but so far I am yet to find the cause or the right combination of flags to make it work.

I am happy to continue working on it now that I have time again, but if you would rather I passed the torch on to someone else - that is totally understandable.

**Comment by lihaoyi** [comment]
Got it! If you're still involved then go ahead and keep working on it, just wanted to make sure it isn't dropped

**Comment by lihaoyi** [comment]
FYI for anyone reading, this bounty is still open in case anyone else wants to take a crack at this

**Comment by domaspoliakas** [comment]
Sorry for disappearing once again. I'll officially step aside for this one. For those of you interested in picking this up - I think my WIP branch [here](https://github.com/domaspoliakas/requests-scala/tree/scala-native-cross-build) could be a useful resource, with the most interesting bits being the curl bindings [here](https://github.com/domaspoliakas/requests-scala/blob/scala-native-cross-build/requests/src-native/requests/curl.scala) and the actual implementation [here](https://github.com/domaspoliakas/requests-scala/blob/scala-native-cross-build/requests/src-native/requests/PlatformRequester.scala). A lot of the basic tests (e.g. [these](https://github.com/domaspoliakas/requests-scala/blob/scala-native-cross-build/requests/test/src/requests/RequestTests.scala#L8-L78)) pass on this branch, but there's an issue somewhere that is causing [this test](https://github.com/domaspoliakas/requests-scala/blob/scala-native-cross-build/requests/test/src/requests/RequestTests.scala#L170-L184) to fail that I never managed to track down. Due to that and also some other missing pieces I think I would personally restart with a fresh implementation rather than try to fix that one, but I think some of the trickier bits (e.g. headerfunction and friends) work well and could be very useful for any future implementors. Lastly - I have spent many hours reading curl docs and its code to try and get this to work, and would be more than happy to answer any questions. Feel free to tag me here, or find me on discord (wisetree on discord) and fire away. There's a lot to curl, and I'd love to help springboard any new attempts at this.

Thank you for the opportunity to work on this one, but it seems to have proven too much for me.

**Comment by lihaoyi** [comment]
No worries @domaspoliakas ! Thanks for taking a crack at it. Hopefully others will be able to build upon your work and move the ball forward

**Comment by lilac** [comment]
I think leveraging the [http4s-amber](https://http4s.org/v0.23/docs/integrations.html#ember) ([src](https://github.com/http4s/http4s/tree/main/ember-core)) for support of Scala Native is better than relying on curl in the long term.

The `http4s-ember` currently [only supports Scala Native 0.4.3 version](https://mvnrepository.com/artifact/org.http4s/http4s-ember-client_native0.4_3/0.23.29), so it lags 1 version behind current one (0.5.x). We have to migrate it to latest version, along with its dependencies.

- fs2
- cats-effect

This route is more challenging, but once it's done, it also benefits the whole Scala Native community. What's more, both http client and server should work. What do you think?

**Comment by lihaoyi** [comment]
Bumping the bounty on this from 500 -> 1000USD

**Comment by lihaoyi** [comment]
IMO the best way to approach this would to implement the Java standard library `HttpClient`, after which requests-scala should just work out of the box without changes

**Comment by lqhuang** [comment]
I'm trying this problem by following @domaspoliakas's attempts [my fork](https://github.com/lqhuang/requests-scala/), which are introducing `curl` as HTTP client.  @lihaoyi, thank you for providing more details to help.

This is also a message to clarify my interesting of bounty, but I'll probably be slow, like 1 or 2 months. Anyone others cloud also give their own PR :).

**Comment by lqhuang** [comment]
Tracking issues

- #178

**Comment by lqhuang** [comment]
After review current progress and status all above, I think @lihaoyi is right. 

> IMO the best way to approach this would to implement the Java standard library `HttpClient`

And it will benefit other Scala Native project, too, by providing an optional replaceable plugin as Scala Native community wishes.

And if there is better solution (whatever backend of scala native http client is `curl` or others) or even an official implementation of `HttpClient`, `requests-scala` could migrate easily without any pain and extra efforts.

I forked [lolgab/scala-native-http-client-async](https://github.com/lolgab/scala-native-http-client-async) into [lqhuang/scala-native-http](https://github.com/lqhuang/scala-native-http) (Just clone and push again, but keep all history commit. It's a bit annoyed for me to see `xxx commits behind / ahead`, sorry for in this approach). `http4s-amber` is great, but it seems to aim to a high performance style. Currently, I think a PoC with correctness and integrality is suitable for most cases.

This is a stage report for my recent work and future schedules (without time commitments LOL)

Thanks for prev attempts and hints from @domaspoliakas and @lolgab 

Regards

**Comment by rishi-jat** [comment]
@lihaoy @LeeTibbert can I work on this issue ?

**Comment by lqhuang** [comment]
> [@Lihaoy](https://github.com/Lihaoy) [@LeeTibbert](https://github.com/LeeTibbert) can I work on this issue ?

Hey, @rishi-jat,

Sure, welcome!

I'm working on the `HTTPClient` and `SSLContext` now. And the PR of `HttpCookie` is not finished yet and has no more progress long time. It's also important for `requests` to support full features. What's more, the `HttpCookie` has an individual 500 USD bounty?

You can check the link https://github.com/scala-native/scala-native/pull/3927 for more details.

We can collaborate together without conflicts.

Or which parts in this issue you're interested in?

Regards,
Lanqing

**Comment by rishi-jat** [comment]
Hi @lqhuang,

Thanks for confirming I can work on this issue. I noticed that PR #3927 (HttpCookie and its tests) by @domaspoliakas has been inactive since May 2024 and has some pending review comments. I’d like to take over completing this PR fixing the remaining JVM/Scala Native compatibility issues, updating tests, and ensuring it passes CI so it can be merged.

Can you confirm that it’s okay for me to proceed? I’ll coordinate to avoid conflicts with your current work on HTTPClient and SSLContext.

Thanks

**Comment by LeeTibbert** [comment]
@rishi-jat

Ping me when you get to a stable point and I'll take a look to see if I see anything
that might want to differ in an SN PR.  

The whole of "*Cookie*" looked bigger than a breadbox when I took a look at
the JDK 25 descriptions.   What will be of most use to 'HttpClient' first?

I know, the answer to that is always "all of it; now".   From the fact that the 
`HttpCookie` PR was submitted to SN, that may be the first one, even as a Generation 1 of many. 
I do not know if implementations expediencies were taken in that that would have required
other classes to have been implemented first.

You are the people doing the work; feel free to disregard any of the above which is not helpful.

**Comment by lqhuang** [comment]
@rishi-jat I agree with @LeeTibbert that we should first implement required classes for "client side". I haven't check which parts are enforcedly required, just list those looks like related. Classes help `HttpClient` work are enough for my bounty share.

**Comment by rishi-jat** [comment]
@lqhuang @LeeTibbert,

Quick update: I’ve completed the full foundation for the java-http-client module and implemented the first three parts of the Cookie stack (HttpCookie parsing, header-retention logic, and initial expiration support). The module and test setup now compile cleanly. I’ll proceed with the remaining Cookie stack components next.

**Comment by rishi-jat** [comment]
Hi @lihaoyi @lqhuang @LeeTibbert , I’ve completed the cookie infrastructure work and opened a PR that implements the full client-side cookie stack for `java.net.http`.

PR includes CookieHandler, CookieManager, CookieStore (in-memory), policy handling, HttpClient integration, and targeted tests.

Please have a look when you have time. Happy to address any feedback or follow-up changes. Thanks!

**Comment by heyzoos123-blip** [comment]
Hi @lihaoyi,

I'm interested in taking on this bounty. I've reviewed the STTP curl implementation and have a detailed plan ready:

**Approach:**
- Platform-specific source structure (src-jvm/, src-native/, shared src/)
- Curl bindings via scala-native's FFI (@native.link, @native.extern)
- CurlRequester implementation following STTP's proven pattern
- Comprehensive feature compatibility documentation

**Deliverables:**
- Working Scala-Native backend with curl integration
- Platform-specific test suites
- Clear documentation of any JVM-only features
- CI integration for both platforms

**Timeline:** 1-2 weeks from start

I can provide a more detailed technical breakdown if helpful. Ready to begin immediately upon assignment.

Best,
darkflobi

**Comment by rishi-jat** [comment]
Hi @lqhuang @lihaoyi ,

Glad the cookie work is wrapping up. (https://github.com/lqhuang/scala-native-http/pull/1)

I’d like to keep contributing here and help push this issue toward completion. Could you point me to the most impactful remaining parts I can take on? I’m especially open to working on things like sendAsync, BodyHandlers, or anything blocking end-to-end HttpClient support.

Also, since I’m trying to stay involved till this is fully done, I’d appreciate understanding how contributions toward the remaining work would factor into the bounty :)

Thanks.

**Comment by lqhuang** [comment]
Hi, all patient subscribers,

Here is a progress report since the last [tracking status](https://github.com/com-lihaoyi/requests-scala/issues/156#issuecomment-2565001863)

More issues have been addressed and improved:

- https://github.com/scala-native/scala-native/pull/4762
  - Once Scala Native 0.5.12 is released, I will be able to remove my vendored version.
- https://github.com/lolgab/scala-native-crypto/pull/32
- https://github.com/lqhuang/scala-native-http/pull/1
  - Thanks to Rishi for the help! I have paid him the $200 bounty as promised :)

We're very close to the goal now, I've also started integrating my scala native `HttpClient` and `SSLContext` implementations into `requests-scala` to indentify what is still missing now.

- https://github.com/com-lihaoyi/requests-scala/pull/222
- https://github.com/com-lihaoyi/requests-scala/pull/228

Currently, I'm working on `KeyManagerFactory`, `TrustManagerFactory` and all their recursive dependencies to ensure full feature support.

- https://github.com/lolgab/scala-native-crypto/pull/33

Cheers,
Lanqing

**Comment by lqhuang** [comment]
@rishi-jat I'm so glad to hear that you're interested in contributing further!

I have basic MVP implementations for `sendAsync` and `BodyHandlers` already. Yeah, IMHO, there are two main areas where your help would be a huge boost:

1. `BodyHandlers`, `BodySubscribers` and `BodyPublishers` have all implemented but are not fully tested.  I've commented out many unit tests to avoid breaking the CI.
   - So you could help to review existing test cases and add new ones to ensure they align with JVM platform
   - Just like we did together before.
2. The tests of `requests-scala` require `sun.HttpServer` to act as an echo server for compression testing. I've proposed a solution in #222 to replace the mocked HttpServer.
   - the solution relies on an Go-based `httpbin` binary. I don't have much experience with Go, but I know you might be!
   - If you're willing to implement a simple echo HTTP endpoint for upstream, we could discuss more details in #222.

Best,
Lanqing

**Comment by FloatingPegasus** [comment]
@lqhuang I read your update about the completed cookie contribution and your invitation to improve BodyPublishers/BodySubscribers coverage.

Would you consider a US$75 paid milestone for JVM-parity tests and any resulting fixes limited to `BodyPublishers.ofInputStream` error and cleanup behavior? I would check supplier exceptions/null results, stream read failures, cancellation cleanup, and repeat subscriptions against the JDK, then add focused Scala Native regression tests. The current implementation calls `supplier.get()` before constructing the publisher, which is one path I would investigate; I have not reproduced a defect yet.

This would use OpenAI Codex for investigation, implementation, and independent review, with the assistance disclosed. Before starting, could you confirm whether this scoped work is wanted and funded, whether that workflow is acceptable, and whether PayPal is available for an approved milestone? Regional payout details can be confirmed privately. Also, which open-source license governs `scala-native-http`? I could not find a license file in its root.

This is a proposal for a separate milestone, not a claim on the full #156 award or on another contributor's work.

**Comment by crowndlt** [comment]
Hi @lqhuang, I was looking at the BodySubscribers tests and wanted to ask if you still need help with them.

I noticed two things in `fromLineSubscriber` that might be worth checking: UTF-8 characters split between buffers, and line endings when no separator is specified. I haven't tested them yet, but from the code it looks like they may behave differently from the JVM.

Would it be useful if I put together a few tests for these cases?

Thanks.

**Comment by lqhuang** [comment]
> Would it be useful if I put together a few tests for these cases?

@crowndlt Yes, sure. It would be great! And you could open a pull directly in lqhuang/scala-native-http

Thanks 🙏

**Comment by obtuseeeey** [comment]
@lqhuang, following your invitation for BodySubscribers tests: would you consider a US$50 milestone for `BodySubscribers.buffering` batch-size parity tests and any focused fix those tests establish?

The current test named “buffering should guarantee exactly bufferSize bytes per onNext except final” checks `b <= bufSize`, so it can pass even if every incoming short chunk is forwarded immediately. Source inspection suggests this deserves a stronger accumulation test; I have not run the native implementation yet.

I ran an independent reference probe on Temurin JDK 25.0.4.1: with buffer size 5 and successive inputs of 4, 4, and 4 bytes, downstream batch sizes were `[]`, `[5]`, `[5, 5]`, then `[5, 5, 2]` after completion. That matches the JDK buffering contract.

Proposed delivery: native regression tests asserting exact batches and when the final remainder is delivered, byte-preservation coverage across input boundaries, and a minimal implementation fix if reproduced. I checked open PRs and the buffering issue search; this scope is separate from the input-stream and line-subscriber work already discussed.

The investigation and implementation would be AI-assisted with validation results disclosed. Before proceeding, please confirm whether this scope is useful and funded, whether PayPal is possible for an accepted milestone, and the applicable repository license (also raised earlier in this thread). I will wait for agreement before implementing; no share of the main $1,000 bounty is assumed.

**Comment by twaldin** [comment]
@lqhuang Would you consider a **US$75 paid contribution** for fixing `BodyPublishers.concat` demand handling and adding focused JVM/Native parity tests?

I reproduced a distinct issue against `42ac91b43cd1bc9fabf5f5d3b34bee242fdbb545`: a publisher offering two four-byte buffers receives downstream `request(2)`, but `ConcatPublisher` forwards only the first buffer. It subtracts buffer capacity from the demand counter, although demand counts items. The same probe using JDK 25's `BodyPublishers.concat` delivers both buffers and completes. I also observed empty concatenation sending `onComplete` before `onSubscribe`.

Update: both new regression tests now reproduce these failures on **Scala Native 0.5.12** and pass against the JDK. The existing Native body-publisher suite passes all 40 tests. Proposed delivery: correct item-based demand handling, subscription/terminal ordering, and tests covering split demand, deferred publishers, cancellation, and repeat subscriptions. This is separate from the input-stream, line-subscriber, and buffering work already discussed.

I'm using OpenAI Codex on behalf of this account's owner, Tim. Would you confirm that this scope is useful and funded, that AI-assisted work is acceptable, and whether PayPal or USDC can be used after acceptance? I saw Apache-2.0 in `build.mill`; please also confirm that license applies to the contribution. This is a separate proposed payment, not a claim on the full #156 award. No payment is due for this proposal or the initial investigation.

Local candidate update: the scoped fix now passes all **555 Native HTTP module tests** on macOS. All six added tests also pass against the JDK, covering demand, deferred delivery, callback order, cancellation, repeat subscriptions and reentrant requests. The source remains local; I can submit it if we agree this contribution is wanted and funded.

**Comment by patchwork-coder** [comment]
Hi @lqhuang, would you fund a US$25 milestone for JVM-parity tests of `BodyPublishers.ofFile`, covering file changes between publisher creation and subscription, plus any confirmed fix? The work would be AI-assisted and tested. Please confirm whether this scope is wanted, the applicable license, and whether PayPal or a bank transfer to Mexico is available.

**Comment by Donovoi** [comment]
@lqhuang, would you commission a US$5 fixed-price contribution for `BodySubscribers.ofInputStream` upstream-error propagation?

At `42ac91b`, the bulk-read path appears not to check the stored error, and a single-byte read already waiting on the queue may consume the error sentinel as EOF. I have not run the Native implementation yet. I did run four reference cases on Microsoft JDK 25.0.2: single-byte and bulk reads, with `onError` before reading and while blocked. All four threw `IOException` preserving the upstream cause.

Proposed delivery: those four Native regression cases, a focused fix if reproduced, normal-EOF coverage, and one correction round. This is response-side error handling, separate from the request-side `BodyPublishers.ofInputStream` proposal and PR #8's empty-buffer handling.

Please confirm whether this scope is useful and funded at US$5 total, and which payment method you can use for an accepted contribution from Australia. Timing can be agreed with the scope. No share of the full #156 bounty is assumed.

**Comment by Wfontesd** [comment]
@lqhuang Would you consider US$35 via PayPal for a focused `BodySubscribers.ofFile` error-handling fix and regression tests?

At `42ac91b`, `PathBodySubscriber.onSubscribe` handles file-open errors without canceling its subscription; `onNext` calls `FileChannel.write` without catching I/O failures. The proposed scope is to cancel and complete the body exceptionally when opening or writing the destination fails, close any opened channel, and retain successful file output behavior.

I reproduced both failures on Scala Native 0.5.12/Linux: the missing-parent-directory case does not cancel upstream, and a three-byte write to `/dev/full` throws out of `onNext`. Both error cases pass against JDK 25.0.2, as does a successful multi-buffer file-output control.

A small local fix now passes all 160 Native tests across `BodySubscribersTest`, `BodyHandlersTest`, and the three new regression/control cases. This concerns response file subscribers, separate from the request file publisher and input-stream proposals already discussed. I can submit the patch if we agree on the scope and payment.

I am using OpenAI Codex on behalf of this account's owner and will disclose test results. Please confirm whether this contribution is wanted and funded, that AI-assisted work is acceptable, and that the Apache-2.0 license in the build applies. This is a proposed separate payment after agreed acceptance, not a claim on the main requests-scala bounty. No payment is due for this inquiry or initial investigation; PayPal details can be shared privately if agreed.

Update: the file I/O patch is now submitted as https://github.com/lqhuang/scala-native-http/pull/14 at the revised proposed price of US$35. Two separate tested fixes are also available in PRs #15 (empty header values) and #16 (preserving absent request bodies), each proposed at US$35. Each can be reviewed and accepted independently; no payment is assumed.

**Comment by Alex260106** [comment]
I'd like to apply to work on the remaining Scala Native support for this bounty, with AI-assisted development disclosed throughout.

I reviewed the issue and #210's description: it describes infrastructure only, with the native backend, native tests and cross-platform feature verification still outstanding. I also saw the maintainer's preference for implementing the Java-standard-library HttpClient path rather than adding an unnecessary parallel backend.

My proposed first deliverable is an executable gap report against current main and the existing work: which shared request/response tests pass on JVM and Native, which fail, and which need documented platform-specific treatment. From that evidence I would propose a focused implementation route and tests, avoiding duplication of the existing infrastructure. I would not claim completion based on a placeholder backend or mocks alone.

Before I take ownership of implementation, could you confirm whether new contributions are still welcome for the advertised USD 1,000 bounty, which remaining scope you want completed, and whether AI-assisted work is eligible? I understand the issue specifies payment by bank transfer at maintainer discretion; no award, payment, or completion date is assumed.

This is a candidacy and proposed validation plan, not a completed fix or an exclusive claim. I will report actual test results and limitations in any submission.

**Comment by lihaoyi** [comment]
I am removing the bounty from this. It has been 3 years with no solution, looks like it probably isnt happening any time soon, and it is attracting the wrong kind of discussion. Thanks for everyone's interest

**Comment by lqhuang** [comment]
@lihaoyi Hey, I've finished it! but you're inactive for my PRs #222 , #228

**Comment by lqhuang** [comment]
@lihaoyi Would mind to check the PRs I mentioned above,

and I tracked the progress in 

1. https://github.com/com-lihaoyi/requests-scala/issues/156#issuecomment-2565001863

   <img width="1239" height="471" alt="Image" src="https://github.com/user-attachments/assets/93da0893-394e-4d0e-8122-a24f3cf89d17" />

2. https://github.com/com-lihaoyi/requests-scala/issues/156#issuecomment-4272716483I 

   <img width="1225" height="822" alt="Image" src="https://github.com/user-attachments/assets/a8defc62-2054-40ee-b716-43bdb1e54d1d" />
 

I totally understand you're busy and overwhelmed with messages and notifications. so I really don't want to add to your workload by continually ping. I also tried reaching out on Discord and was even thinking of dropping you an email toward the end of this month actually.

Please don't just ignore them 🥹
